File: Programmers/Python/Level 2/n^2 배열 자르기/solution.py
# https://programmers.co.kr/learn/courses/30/lessons/87390

def solution(n, left, right):
    answer = []
    for num in range(left, right + 1):
        # 1차원 배열로 놨을때 각 위치는 인덱스를 n으로 나눈 (몫, 나머지)
        answer.append(max(divmod(num, n)) + 1)
    return answer

File: Programmers/Javascript/Level 2/n^2 배열 자르기/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/87390?language=javascript

function solution(n, left, right) {
  const answer = [];

  for (let idx = left; idx <= right; idx++) {
    // Math.floor(idx / n)는 행, idx % n은 열 번호
    answer.push(Math.max(Math.floor(idx / n), idx % n) + 1);
  }
  return answer;
}
